specification/quantum/Quantum.Workspace/common/base-operations.tsp

```diff
@@ -0,0 +1,85 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+import "./common.tsp";
+
+using OpenAPI;
+using TypeSpec.Versioning;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Rest.Resource;
+using Azure.Core;
+using Azure.Core.Traits;
+
+alias DefaultServiceTraits = NoRepeatableRequests &
+  NoConditionalRequests &
+  NoClientRequestId;
+alias AzureCoreOperations = Azure.Core.ResourceOperations<DefaultServiceTraits>;
+
+@useDependency(DefaultAzureCoreVersion)
+namespace Quantum.Workspace.BaseOperations {
+  alias ResourceOkOrNoContentResponse<Resource extends TypeSpec.Reflection.Model> = Azure.Core.Foundations.ResourceOkResponse<Resource> | TypeSpec.Http.NoContentResponse;
+
+  @doc("Generic POST operation.")
+  @Foundations.Private.ensureVerb("PostOperation", "POST")
+  @post
+  op PostOperation<
+    TRequest extends TypeSpec.Reflection.Model,
+    TResponse extends TypeSpec.Reflection.Model
+  > is Azure.Core.Foundations.Operation<
+    {
+      ...CommonParams;
+      ...TRequest;
+    },
+    TResponse
+  >;
+
+  @doc("Operation to create a resource.")
+  op CreateOperation<TResource extends TypeSpec.Reflection.Model> is AzureCoreOperations.ResourceCreateOrReplace<TResource>;
+
+  @doc("Operation to patch a resource.")
+  @updatesResource(Resource)
+  @patch
+  @Azure.Core.Foundations.Private.ensureVerb("PatchOperation", "PATCH")
+  op PatchOperation<Resource extends TypeSpec.Reflection.Model> is Azure.Core.Foundations.ResourceOperation<
```

Comment:
Check with @allenjzhang if there is a better way to reuse the Azure.Core Patch operation.

Reply:
Allen, Johan and I discussed about the need for this custom operation.
Currently there is no support in *TypeSpec* for this scenario.
We agreed to open an Issue in the *TypeSpec Azure repo* (https://github.com/Azure/typespec-azure/issues/475) and accept this implementation.

Reply:
Discussed in meeting.